src/com/mendix/widget/badge/Badge.ts

```diff
@@ -0,0 +1,113 @@
+import * as dojoDeclare from "dojo/_base/declare";
+import * as WidgetBase from "mxui/widget/_WidgetBase";
+
+import { createElement } from "react";
+import { render } from "react-dom";
+
+import { Badge as BadgeComponent } from "./components/Badge";
+import { BadgeButton } from "./components/BadgeButton";
+import { BadgeLabel } from "./components/BadgeLabel";
+
+export interface OnClickProps {
+    microflow?: string;
+    guid?: string;
+    applyto?: string;
+}
+
+class Badge extends WidgetBase {
+    // Attributes from modeler
+    private attrValue: string;
+    private attrStyle: string;
+    private attrLabel: string;
+    private badgeType: "btn" | "label" | "badge";
+    private label: string;
+    private badgeClass: string;
+    private onclickMicroflow: string;
+    // Internal variables
+    private contextObject: mendix.lib.MxObject;
+
+    postCreate() {
+        this.updateRendering();
+    }
+
+    update(object: mendix.lib.MxObject, callback: Function) {
+        this.contextObject = object;
+        this.resetSubscriptions();
+        this.updateRendering();
+
+        callback();
+    }
+
+    private updateRendering() {
+        const BadgeElement = this.badgeType === "btn" ? BadgeButton
+            : this.badgeType === "label" ? BadgeLabel
+            : BadgeComponent;
+
+        const clickAble = this.contextObject && this.contextObject.getGuid() && this.onclickMicroflow;
+
+        render(createElement(BadgeElement as any, {
+            badgeValue: this.getValue(this.attrValue, ""),
+            disabled: this.contextObject ? undefined : "disabled",
```

Comment:
I think the mendix widgets use string type

<button type="button" class="btn mx-button mx-name-microflowButton3 btn-default" id="mxui_widget_ActionButton_1" data-mendix-id="39_11" tabindex="0" widgetid="mxui_widget_ActionButton_1" disabled="disabled"> Refresh </button>